### PR TITLE
Allow custom ID for document store; allow document generation on cache miss via block.

### PR DESCRIPTION
**Pull request by victorluft**

### CODE REVIEW

**Comment by bklang** [comment]
Tests?

File: lib/virginia/document_cache.rb [hunk 1]
```diff
@@ -50,10 +51,19 @@ def delete(id)
 
     # Retrieves a document from the cache
     # @param [String] id ID of the document to be retrieved from the cache
+    # @param [Fixnum, Nil] lifetime The amount of time in seconds the document should be kept. If nil, document will be kept indefinitely.
+    # @yield If given, will be used to generate the document, store it, and then return.
     # @return [Object] document Returns the document if found in the cache
     # @raises [NotFound] If the document is not found in the cache
-    def fetch(id)
-      raise NotFound unless @documents.has_key? id
+    def fetch(id, lifetime = 10)
```
Comment (bklang) [inline lib/virginia/document_cache.rb]: Since we're using this twice, make `lifetime`'s default value a constant?

**Comment by bklang** [comment]
And then I just realized I asked for tests and there aren't any to begin with.  But still, tests would be nice.

File: lib/virginia/document_cache.rb (hunk 1 above)
Comment (victorluft) [inline lib/virginia/document_cache.rb]: Yup. Probably want to also increase default value, as this is in seconds and minimum resolution of cache sweep is in minutes.

**Pull request merged**